lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -406,6 +407,25 @@ function (array $fieldMapping) use ($class): ?string {
                             );
                         }
 
+                        if (
+                            isset($fieldMapping['enumType'])
+                            && $propertyType !== $fieldMapping['enumType']
+                            && interface_exists($propertyType)
+                            && is_a($fieldMapping['enumType'], $propertyType, true)
+                        ) {
+                            if ($metadataFieldType === (string) (new ReflectionEnum($fieldMapping['enumType']))->getBackingType()) {
```

Comment:
You're basing this on a comparison between `$metadataFieldType` and the backing type of the enum, yet only one of those appears in the error message.